gr-digital/lib/crc16_async_bb_impl.h

```diff
@@ -0,0 +1,48 @@
+/* -*- c++ -*- */
+/*
+ * Copyright 2021 Cesar Martinez.
+ *
+ * This file is part of GNU Radio
+ *
+ * SPDX-License-Identifier: GPL-3.0-or-later
+ * 
+ */
+
+#ifndef INCLUDED_DIGITAL_CRC16_ASYNC_BB_IMPL_H
+#define INCLUDED_DIGITAL_CRC16_ASYNC_BB_IMPL_H
+
+#include <gnuradio/digital/crc16_async_bb.h>
+#include <boost/crc.hpp>
+
+namespace gr {
+namespace digital {
+
+class crc16_async_bb_impl : public crc16_async_bb
+{
+private:
+    
+    boost::crc_optimal<16, 0x1021, 0xFFFF, 0, false, false> d_crc_ccitt_impl;
```

Comment:
I don't know anything about this stuff, but [this link](http://stackoverflow.com/questions/17196743/ddg#17196947) suggests that either you should start with 0x0000 or this isn't really CCITT?

Reply:
In my [favourite online CRC calculator](http://www.sunshine2k.de/coding/javascript/crc/crc_js.html), you can see that there are two flavours of CRC-16-CCITT. What is called there CCITT_ZERO, which starts with `0x0000`, and CCITT_FALSE, which starts with `0xffff`.

To make things more complicated, the CRC-16-CCITT used by AX.25 is also different. It has a final `0xffff` XOR value and reflects the input and output.

It seems that the intention here was to implement CCITT_FALSE, which I think it's more popular than CCITT_ZERO. Looking at #2366, the idea was to implement the CRC used by STANAG 4202, and Yaesu System Fusion, but I don't know what flavour these use.